### PR TITLE
poe.ninja support

### PR DESCRIPTION
**Pull request by kai-oswald**
Adjusted for the new architecture.
I had to reinstall Newtonsoft.Json tho, since the poe.trade API is inconsistent with its response naming and I needed `JsonProperty`.

Feel free to adjust, but this should be a good base.
What's still missing is a custom refresh, which could be placed into the TrayIcon.

See #5

### CODE REVIEW

**Comment by JadedCricket** [comment]
We are trying to move away from newtonsoft and towards Microsoft’s implementation (System.Text.Json). They have an attribute JsonPropertyName that should achieve the same result

File: src/Sidekick.Business/Apis/PoeNinja/PoeNinjaClient.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public async Task<PoeNinjaQueryResult<PoeNinjaItem>> GetItemOverview(string leag
 
             var responseString = await PerformRequestAndValidateResponse(url);
 
-            return JsonConvert.DeserializeObject<PoeNinjaQueryResult<PoeNinjaItem>>(responseString);
+            return JsonSerializer.Deserialize<PoeNinjaQueryResult<PoeNinjaItem>>(responseString);
```
Comment (leMicin) [inline src/Sidekick.Business/Apis/PoeNinja/PoeNinjaClient.cs]: Use DeserializeAsync for better performance

File: src/Sidekick.Business/Apis/PoeNinja/PoeNinjaClient.cs [hunk 2]
```diff
@@ -39,7 +39,7 @@ public async Task<PoeNinjaQueryResult<PoeNinjaCurrency>> GetCurrencyOverview(str
 
             var responseString = await PerformRequestAndValidateResponse(url);
 
-            return JsonConvert.DeserializeObject<PoeNinjaQueryResult<PoeNinjaCurrency>>(responseString);
+            return JsonSerializer.Deserialize<PoeNinjaQueryResult<PoeNinjaCurrency>>(responseString);
```
Comment (leMicin) [inline src/Sidekick.Business/Apis/PoeNinja/PoeNinjaClient.cs]: Use DeserializeAsync for better performance

**Comment by domialex** [comment]
For me it's good, just that I would rather make a separate model for the overlay results instead of using the one from the PoeApi, be more MVC, but I think we can do that later.
Also, we will need in the future to make the overlay pop AS SOON AS we hotkey for a price check IF the parsed item is parsable, then we populate poe.ninja/poeprices.info and poeApi results.

**Comment by domialex** [comment]
Related to #140.

**Comment by domialex** [comment]
Related to #141.

**Pull request merged**